### PR TITLE
Added Channel#banners, Channel#keywords, and Channel#featuredChannels

### PR DESCRIPTION
**Pull request by Tenpi**
I added the endpoint `/channels?part=brandingSettings` for channels, and properties `banners`, `keywords`, and `featuredChannels` which all come from brandingSettings. 

I also defined a type for Banners.
```ts
export type Banners = {
  bannerImageUrl: string
  bannerMobileImageUrl: string
  bannerMobileLowImageUrl: string
  bannerMobileMediumImageUrl: string
  bannerMobileHdImageUrl: string
  bannerMobileExtraHdImageUrl: string
  bannerTabletLowImageUrl: string
  bannerTabletImageUrl: string
  bannerTabletHdImageUrl: string
  bannerTabletExtraHdImageUrl: string
  bannerTvImageUrl: string
  bannerTvLowImageUrl: string
  bannerTvMediumImageUrl: string
  bannerTvHighImageUrl: string
}
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "test": "mocha --exit --require ts-node/register --require source-map-support/register \"./test/!(oauth).spec.ts\"",
     "coverage": "nyc yarn test && nyc report --reporter=text-lcov | coveralls",
     "docs": "gulp typedoc",
-    "prepublishOnly": "gulp"
+    "prepublishOnly": "gulp",
+    "temp": "tsc && node out/temp.js"
```
Comment (jasonhaxstuff) [inline package.json]: Remove this

File: package.json [hunk 2]
```diff
@@ -30,6 +31,7 @@
     "@types/node": "12.6.2",
     "chai": "^4.1.2",
     "coveralls": "^3.0.5",
+    "dotenv": "^8.1.0",
```
Comment (jasonhaxstuff) [inline package.json]: What's this for?

File: yarn.lock [hunk 3]
```diff
@@ -984,6 +984,11 @@ doctrine@0.7.2:
     esutils "^1.1.6"
     isarray "0.0.1"
 
+dotenv@^8.1.0:
+  version "8.1.0"
+  resolved "https://registry.yarnpkg.com/dotenv/-/dotenv-8.1.0.tgz#d811e178652bfb8a1e593c6dd704ec7e90d85ea2"
+  integrity sha512-GUE3gqcDCaMltj2++g6bRQ5rBJWtkWTmqmD0fo1RnnMuUqHNCt2oTPeDnS9n6fKYvlhn7AeBkb38lymBtWBQdA==
+
```
Comment (jasonhaxstuff) [inline yarn.lock]: See previous comment

File: src/entities/channel.ts [hunk 4]
```diff
@@ -140,6 +155,9 @@ export class Channel {
     this.name = data.snippet.title
     this.about = data.snippet.description
     this.full = data.kind === 'youtube#channel'
+    this.banners = data.brandingSettings.image
+    this.keywords = data.brandingSettings.channel.keywords
+    this.featuredChannels = data.brandingSettings.channel.featuredChannelsUrls.map((c) => `https://www.youtube.com/channel/${c}`)
```
Comment (jasonhaxstuff) [inline src/entities/channel.ts]: Remove parenthesis around `c` and rename `c` to `url`

File: src/entities/channel.ts [hunk 5]
```diff
@@ -140,6 +155,9 @@ export class Channel {
     this.name = data.snippet.title
     this.about = data.snippet.description
     this.full = data.kind === 'youtube#channel'
+    this.banners = data.brandingSettings.image
+    this.keywords = data.brandingSettings.channel.keywords
```
Comment (jasonhaxstuff) [inline src/entities/channel.ts]: `.split(' ')` this and make `keywords` a `string[]`

File: src/entities/channel.ts [hunk 6]
```diff
@@ -106,6 +116,11 @@ export class Channel {
    */
   public comments: YTComment[]
 
+  /**
+   * The urls of all the channels this channel features.
```
Comment (jasonhaxstuff) [inline src/entities/channel.ts]: Change urls to URLs

File: .gitignore [hunk 7]
```diff
@@ -8,3 +8,5 @@ typings/
 .nyc_output/
 *.log
 .coveralls.yml
+.env
+src/temp.ts
```
Comment (jasonhaxstuff) [inline .gitignore]: Remove this.

File: package.json (hunk 2 above)
Comment (jasonhaxstuff) [reply package.json]: Ignore this, I just saw your other PR

**Comment by Tenpi** [comment]
Thanks, it's possible for `brandingSettings` to be undefined so I needed to wrap everything in ternary statements. They can also be empty if the channel have any banners, keywords, or featured channels set.

**Comment by jasonhaxstuff** [comment]
@Tenpi - It'll be around 8 hours until I can merge this since I have to wait for my YouTube API quota and I'm about to head to bed. It looks good though, I just need to get the test coverage.

**Pull request merged**